src/main/java/org/spongepowered/api/command/parameter/Parameter.java

```diff
@@ -0,0 +1,1236 @@
+     *
+     * @param subcommand The {@link Command} to execute
+     * @param alias The first alias of the subcommand
+     * @param aliases Subsequent aliases, if any
+     * @return The {@link Subcommand} for use in a {@link Parameter} chain
+     */
+    static Subcommand subcommand(Command subcommand, String alias, String... aliases) {
+        Subcommand.Builder builder = Sponge.getRegistry()
+                .createBuilder(Subcommand.Builder.class)
+                .setSubcommand(subcommand)
+                .alias(alias);
+        for (String a : aliases) {
+            builder.alias(a);
+        }
+
+        return builder.build();
+    }
+
+    /**
+     * Returns a {@link Parameter.FirstOfBuilder} that allows plugins to attempt
+     * to parse an argument using the supplied parameters in order. Once a
+     * parameter has parsed the argument successfully, no more parameters
+     * supplied here will be attempted.
+     *
+     * @param parameter The first {@link Parameter}
+     * @return The {@link Parameter.FirstOfBuilder} to continue chaining
+     */
+    static Parameter.FirstOfBuilder firstOf(Parameter parameter) {
+        return Sponge.getRegistry().createBuilder(FirstOfBuilder.class).or(parameter);
+    }
+
+    /**
+     * Returns a {@link Parameter} that attempts to parse an argument using the
+     * supplied parameters in order. Once a parameter has parsed the argument
+     * successfully, no more parameters supplied here will be attempted.
+     *
+     * @param first The first {@link Parameter} that should be used for parsing
+     * @param second The second {@link Parameter} that should be used for
+     *               parsing, should the first {@link Parameter} fail to do so
+     * @param parameters The remaining {@link Parameter}s
+     * @return The {@link Parameter}
+     */
+    static Parameter firstOf(Parameter first, Parameter second, Parameter... parameters) {
+        return Sponge.getRegistry().createBuilder(FirstOfBuilder.class).or(first).or(second).orFirstOf(parameters).build();
+    }
+
+    /**
+     * Returns a {@link Parameter} that attempts to parse an argument using the
+     * supplied parameters in order. Once a parameter has parsed the argument
+     * successfully, no more parameters supplied here will be attempted.
+     *
+     * @param parameters The {@link Parameter}s
+     * @return The {@link Parameter}
+     */
+    static Parameter firstOf(Iterable<Parameter> parameters) {
+        return Sponge.getRegistry().createBuilder(FirstOfBuilder.class).orFirstOf(parameters).build();
+    }
+
+    /**
+     * Returns a {@link Parameter.SequenceBuilder} that parses arguments using
+     * the supplied parameters in order.
+     *
+     * @param parameter The first {@link Parameter} in the sequence
+     * @return The {@link Parameter.SequenceBuilder}, to continue building the
+     *         chain
+     */
+    static Parameter.SequenceBuilder seq(Parameter parameter) {
```

Comment:
Why abbreviate the method name?

Reply:
Hangover from API 7, but `sequence` doesn't sit _completely_ correctly with me. Names are welcome for what this is.